oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/AccessLogAnalyzer.java

```diff
@@ -78,10 +79,17 @@ static class Result {
         /**
          * The analyzed metrics result.
          */
-        private ServiceMeshMetrics.Builder metrics;
+        @Builder.Default
+        private ServiceMeshMetrics.Builder metrics = ServiceMeshMetrics.newBuilder();
+
+        @Accessors(fluent = true)
+        private boolean hasDownstreamMetrics;
+        @Accessors(fluent = true)
+        private boolean hasUpstreamMetrics;
 
         public boolean hasResult() {
-            return metrics != null && (metrics.getHttpMetrics().getMetricsCount() > 0 || metrics.getTcpMetrics().getMetricsCount() > 0);
+            return metrics.getHttpMetrics().getMetricsCount() > 0
+                || metrics.getTcpMetrics().getMetricsCount() > 0;
```

Comment:
Should this be `hasDownstreamMetrics || hasUpstreamMetrics`?

Reply:
This is for sidecar analysis, for logs from sidecar. It doesn't care about upstream or downstream. It just cares about whether there is matrics or not.

Reply:
But I can do it to unify the condition to check whether there is metrics

Reply:
Meanwhile, do `hasDownstreamMetrics`/`hasUpstreamMetrics` have meanings when do sidecar analysis?
I am trying to get variable more accurate.

Reply:
Yes, you can make sense of the names: for server sidecar there is always only downstream metrics, for client sidecar there is always only upstream metrics.